### ISSUE TITLE
Cannot complete imports if the package has objects split across multiple files

### ISSUE THREAD

**Issue by KaranAhlawat** [scalameta/metals#5774]
### Describe the bug

I have a package `shop.domain`, which is reflected by the folder structure `shop/domain/`.
Inside `domain`, I have multiple files. Each file has the package as `shop.domain`, not `shop.domain.<filename>`, and each file contains one object. If I go into a another file (same package OR different package), and I start typing `import shop.domain.`, I do not get the expected completions for after the `.`. Is this expected behavior?

![image](https://github.com/scalameta/metals/assets/56113511/815c24e2-b8a5-40d8-8b42-74f26c7e7dce)
![image](https://github.com/scalameta/metals/assets/56113511/84636711-5dd2-4273-a656-b34d9312ad01)
![image](https://github.com/scalameta/metals/assets/56113511/c9ea17bf-cc54-4938-a914-db8cf529ab58)

### Expected behavior

I expect the `.` to popup all the objects that are defined in that package, across the various files.

### Operating system

Linux

### Editor/Extension

VS Code

### Version of Metals

v1.1.0

### Extra context or search terms

It obviously works if I assign each file to a different package, so `shop/domain/brand.scala` would be `package shop.domain.brand`. And then the imports work fine.

I'm using a mill project, and can try to create a minimum reproducible example if needed.

**Comment by jkciesluk** [comment]
Hi, thanks for reporting! I'm unable to reproduce the issue, so it would be great if you can provide a reproduction.

**Comment by KaranAhlawat** [comment]
Hey, this is what I came up with for the repro

https://github.com/KaranAhlawat/mill-metals-repro

Hopefully this helps, you can try and play around with the `import` statement(s) at the top of the `Main.scala` file to see the autocompletion behavior

**Comment by jkciesluk** [comment]
Thanks! Are you sure your build is correctly imported? Could you run command `Metals: Run Doctor` and see if any build targets are shown? If not, you can try command `Metals: Clear caches and reimport project`.

**Comment by KaranAhlawat** [comment]
Yes, I just checked, the only missing functionality seems to be debugging.
![image](https://github.com/scalameta/metals/assets/56113511/80f4e215-2e7d-496d-ab44-2e1390bd4bc7)
Both modules also seem to be appearing correctly.

I also ran the command `Metals: Clear caches and reimport project` with no difference in the behavior afterwards.

**Comment by jkciesluk** [comment]
It seems to be an issue with mill bsp, I will look into it. For now you can switch to bloop, where it seems to work fine

**Comment by KaranAhlawat** [comment]
Aha! It is indeed, ty. Please let me know if I can help out in some way :) Till then I'll stick to using bloop 😞
